src/stan_math_backend/Stan_math_code_gen.ml

```diff
@@ -472,7 +472,73 @@ using stan::math::lgamma;
 using stan::model::prob_grad;
 using namespace stan::math; |}
 
+let rec xform_readdata sizes s =
+  let get_single_size = function
```

Comment:
Shouldn't this sort of transformation happen in AST_to_mir?

It also seems a little bit hacky, to the untrained eye. Not that I have a better alternative to propose.

Reply:
We talked about this in the meeting - I think we should continue to think about where the line is for this stuff. I think this one in particular is super C++/StanMath dependent and it belongs here, but we might actually want to move more of Ast_to_Mir into the code gen side of things and spin off a separate MIR to MIR translation as part of each backend (that we could hopefully then re-run the optimizer on, or only optimize afterwards).

Reply:
Sounds good!